confirmation_rule.py

```diff
@@ -181,6 +182,44 @@ def is_one_confirmed(store: Store, block_root: Root) -> bool:
     )
 
 
+def is_chain_one_confirmed(store: Store, start_root: Root, end_root: Root) -> bool:
+    """
+    Checks if ``is_one_confirmed(store, block_root)`` returns ``True`` for each block
+    starting from ``end_root`` inclusive down to ``start_root`` exclusive.
+    Returns ``True`` only if each block passes the check.
+    """
+    block_root = end_root
+    while get_block_slot(block_root) > get_block_slot(start_root):
+        if is_one_confirmed(store, block_root):
+            block_root = store.blocks[block_root].parent_root
+        else:
+            break
+
+    if block_root == start_root:
+        return True
+    else:
+        return False
```

Comment:
nit: maybe a bit more compact and clear 
```suggestion
    block_root = end_root
    while block_root != start_root:
        if not is_one_confirmed(store, block_root):
            return False
        block_root = store.blocks[block_root].parent_root
    return True
```